client/gutenberg/editor/components/post-publish-button-or-toggle/index.jsx

```diff
@@ -0,0 +1,225 @@
+/** @format */
+/**
+ * External Dependencies
+ */
+import React, { Component, Fragment } from 'react';
+import { get } from 'lodash';
+import { translate } from 'i18n-calypso';
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies.
+ */
+import { compose, withState } from '@wordpress/compose';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { PostPublishButton } from '@wordpress/editor';
+import { withViewportMatch } from '@wordpress/viewport';
+
+/**
+ * Internal dependencies
+ */
+import VerifyEmailDialog from 'components/email-verification/email-verification-dialog';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { isCurrentUserEmailVerified } from 'state/current-user/selectors';
+import isUnlaunchedSite from 'state/selectors/is-unlaunched-site';
+import isVipSite from 'state/selectors/is-vip-site';
+
+export class PostPublishButtonOrToggle extends Component {
+	componentDidMount() {
+		this.togglePostSaving();
+	}
+
+	componentDidUpdate( prevProps ) {
+		const { userNeedsVerification, verificationNoticeLabel } = this.props;
+		if (
+			userNeedsVerification !== prevProps.userNeedsVerification ||
+			verificationNoticeLabel !== prevProps.verificationNoticeLabel
```

Comment:
I'm wondering why do we need to check if `verificationNoticeLabel` changed—or, in other words, [if the post "status" changed](https://github.com/Automattic/wp-calypso/pull/29594/files#diff-f4811af222e472a6d2de60560f7624b7R169) between `isPublished`, `isBeingScheduled`, or otherwise.

Wouldn't be enough for our purpose to simply disable saving when `userNeedsVerification`, and enable it when the user doesn't?

Reply:
It's needed because `togglePostSaving` (to be changed to `toggleLockPostSaving`) does 2 things:
* Locks/unlocks the post saving
* Creates a notice with a message which is based on the post status

If we don't check the post status, the notice won't change from "To publish, check your email and confirm your address." to "To schedule, check your email and confirm your address." when scheduling the post.

Reply:
Makes sense 👍